packages/react-client/src/ReactFlightClient.js

```diff
@@ -472,30 +472,34 @@ function createElement(
   key: mixed,
   props: mixed,
 ): React$Element<any> {
-  const element: any = enableRefAsProp
-    ? {
-        // This tag allows us to uniquely identify this as a React Element
-        $$typeof: REACT_ELEMENT_TYPE,
-
-        // Built-in properties that belong on the element
-        type,
-        key,
-        props,
-
-        // Record the component responsible for creating this element.
-        _owner: null,
-      }
-    : {
-        // Old behavior. When enableRefAsProp is off, `ref` is an extra field.
-        ref: null,
-
-        // Everything else is the same.
-        $$typeof: REACT_ELEMENT_TYPE,
-        type,
-        key,
-        props,
-        _owner: null,
-      };
+  let element: any;
+  if (__DEV__ && enableRefAsProp) {
+    // `ref` is non-enumerable in dev
+    element = ({
+      $$typeof: REACT_ELEMENT_TYPE,
+      type,
+      key,
+      props,
+      _owner: null,
+    }: any);
+    Object.defineProperty(element, 'ref', {
```

Comment:
Not sure if it helps but it might be worth keeping these as getters so that the object have the same shape whether it's from Flight or JSX.